_posts/linda_xiang/2016-12-07-Cloud-data-transfer.md

```diff
@@ -0,0 +1,191 @@
+
+While migrating the tremendous amount of data generated by NGS among those clouds, it is more efficient and realistic to use as many computers as needed to do the data migration parallelly. Given the big size of each file, it is already a lot to handle even using just a single node. When multiple machines are doing the jobs at the same time, things can quickly get complicated and tough. 
+
+Therefore, a real-time management system is needed to synchronize all jobs status which are done by different machines. We had to come up with a simple, yet flexible, efficient and intelligent way to manage thwhole workflow.
+
+
+## System design and implementation
+
+Here we propose a new solution combines the extremely popular [Git](https://git-scm.com/) and [Elasticsearch](https://www.elastic.co/), giving you a very flexible approach that will work with data of any size and amount. 
+
+Let's see what compoments we will be working with today:
+
+<figure>
+    <img src="{{site.urlimg}}linda_xiang/data-transfer/data_migrate_block.png" />
+    <figcaption>Figure 1. Data Migration Management System Blockgram</figcaption>
+</figure>
+
+* Source: Where the big genomic data files will be downloaded from.
+* Destination: Where the big genomic data files will be uploaded to.
+* Elasticsearch Index: Parse the metadata of data source, generate the JSON format metadata documents and build the Elasticsearch index.
+* Jobs Generator: Query the elasticsearch to generate jobs in JSON which contain the important metadata information about the files to be transferred.
+* Jobs Repo (Git): Where your jobs live. We can use public space like [GitHub](https://github.com/) or self-hosting git repository to store jobs.
+* Cloud Instance: Fetches and moves the job from and among folders in Git repo, downloads the file from the given source and uploads the file to the given destination. 
+
+
+## How it works
+
+Here I will give a concrete example to show you how to migrate PCAWG data from data source([PCAWG in GNOS](http://pancancer.info/)) to data destination([Collaboratory](https://www.cancercollaboratory.org/)). Following is the
+
+<figure>
+    <img src="{{site.urlimg}}linda_xiang/data-transfer/data_transfer.png" />
+    <figcaption>Figure 2. PCAWG Data Migration From GNOS to Collaboratory</figcaption>
+</figure>
+
+#### Data Source: [PCAWG in GNOS](http://pancancer.info/)
```

Comment:
Maybe this, and subsequent sections should be H3 (###) instead of H4 (####) since parent is H2